components/push/android/src/main/java/mozilla/appservices/push/PushManager.kt

```diff
@@ -228,24 +228,81 @@ class DispatchInfo constructor (
 
 /**
  * An API for interacting with Push.
+
+    Usage:
+
+    The push component is designed to be as light weight as possible. The "Push Manager"
+    handles subscription management and message decryption.
+
+    In general, usage would consist of calling:
+
+    ```kotlin
+    val manager = PushManager(
+        senderId = "SomeSenderIDValue",
+        bridgeType = BridgeType.FCM,
+        registrationId = systemProvidedRegistrationValue,
+        databasePath = "/path/to/database.sql"
+    )
+    val newEndpoints = manager.verifyConnection()
+    if newEndpoints.length() > 0 {
+        for (channelId in newEndpoints.keys()) {
+            // send the endpoint (newEndpoint[channelId]) to the process tied to channelId
+        }
+    }
+
+    // On new message:
+    // A new incoming message generally has the following format:
+    // {"chid": ChannelID, "body": Body, "con": Encoding, "enc": Salt, "crypto_key": DH}
+
+    val decryptedMessage = manager.decrypt(
+        channelID=message["chid"],
+        body=message["body"],
+        encoding=message["con"],
+        salt=message.getOrElse("enc", ""),
+        dh=message.getOrElse("crypto-key", "")
+    )
+
+    // On new subscription:
+    val subscriptionInfo = manager.subscribe(channelID, scope)
+
+    // channelID is a UUID4 value that can either be created before hand, or an empty string
+    //           can be passed in and one will be created for you.
+    // scope     is the site scope string. This will be used for rate limiting
+    //
+    // The subscription info matches what is usually passed on to
+    // the requesting application.
+    // This could be JSON encoded and returned.
+
+    // On deleting a subscription:
+    manger.unsubscribe(channelID)
+    // returns true/false on server unsubscribe request. A False may cause a
+    // verifyConnection() failure and new endpoints generation
+
+    // On a new native OS registration ID change:
+    manager.update(newSubscriptionID)
+    // sets the new registration ID (sender ID) on the server. Returns a false if this
+    // operation fails. A failure may prevent future messages from being received.
+
+```
  */
-interface PushAPI {
+interface PushAPI : java.lang.AutoCloseable {
     /**
      * Get the Subscription Info block
      *
-     * @param channelID Channel ID (UUID) for new subscription
-     * @return a Subscription Info structure
+     * @param channelID Channel ID (UUID4) for new subscription, either pre-generated or "" and one will be created.
+     * @param scope Site scope string.
+     * @return a SubscriptionInfo structure
      */
     fun subscribe(
-        channelID: String,
-        scope: String
+        channelID: String = "",
+        scope:
```

Comment:
This looks wrong

Reply:
I'm not quite sure about Kotlin. This compiles cleanly and appears to do what is expected. I'm more than happy to use some other way to specify the default value if there is one.

Reply:
It might be why`ktlint` is failing, shouldn't you type the parameter?

Reply:
ah, didn't see that the scope parameter fell off. Thought you were talking about the channelID default parameter. fixed.